### PR TITLE
IBX-2429: Example of UDW extensibility

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2429
| Versions      | master, 4.0+

### CODE REVIEW

File: code_samples/back_office/udw/assets/js/image-tab/components/images.list.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useState, useContext, useEffect } from 'react';
 import Image from './image';
 import { getImages } from '../services/images.service';
 
-import { RestInfoContext } from '../../../../vendor/ezsystems/ezplatform-admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/universal.discovery.module';
+import { RestInfoContext } from '../../../../vendor/ibexa/admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/universal.discovery.module';
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/components/images.list.js:5]: Now you can use just:
```suggestion
import { RestInfoContext } from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/universal.discovery.module';
```

File: code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
-        return (
-            <div className="m-image-tab">
-                <Tab isContentOnTheFlyDisabled={true} isSortSwitcherDisabled={true} isViewSwitcherDisabled={true}>
-                    <ImagesList />
-                </Tab>
-            </div>
-        );
-    };
+
+import Tab from '../../../vendor/ibexa/admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/components/tab/tab';
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js:3]:
```suggestion
import Tab from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/components/tab/tab';
```

File: code_samples/back_office/udw/assets/js/image-tab/components/images.list.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ const ImagesList = () => {
         return (
             <div {...attrs}>
                 <svg className="ibexa-icon">
-                    <use xlinkHref="/bundles/ibexaplatformicons/img/all-icons.svg#caret-back"></use>
+                    <use xlinkHref="/bundles/ibexaadminui/img/ibexa-icons.svg#caret-back"></use>
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/components/images.list.js:62]:
```suggestion
                    <use xlinkHref={window.ibexa.helpers.icon.getIconPath('caret-back')}></use>
```

File: code_samples/back_office/udw/assets/js/image-tab/components/images.list.js [hunk 4]
```diff
@@ -76,8 +76,8 @@ const ImagesList = () => {
 
         return (
             <div {...attrs}>
-                <svg className="ibexa-icon">
-                    <use xlinkHref="/bundles/ibexaplatformicons/img/all-icons.svg#caret-next"></use>
+                <svg className="ez-icon">
+                    <use xlinkHref="/bundles/ibexaadminui/img/ibexa-icons.svg#caret-next"></use>
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/components/images.list.js:80]:
```suggestion
                    <use xlinkHref={window.ibexa.helpers.icon.getIconPath('caret-next')}></use>
```

File: docs/extending/add_udw_tab.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+<details class="tip">
+<summary>Complete image.tab.module.js code</summary>
+```js
+[[= include_file('code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js') =]]
+```
+</details>
+
+## Add tab to webpack config
+
+In `webpack.config.js`, add the following key:
```
Comment (dew326) [inline docs/extending/add_udw_tab.md:42]:
```suggestion
In `webpack.config.js`, add the following declaration:
```

File: docs/extending/add_udw_tab.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+??? tip "Complete `webpack.config.js` code"
+
+    ```js
+    const Encore = require('@symfony/webpack-encore');
+    const path = require('path');
+    const getIbexaConfig = require('./ibexa.webpack.config.js');
+    const ibexaConfig = getIbexaConfig(Encore);
+    const customConfigs = require('./ibexa.webpack.custom.configs.js');
+    const ibexaConfigManager = require('./ibexa.webpack.config.manager.js');
+    Encore.reset();
```
Comment (dew326) [inline docs/extending/add_udw_tab.md:126]:
```suggestion

    Encore.reset();
```

File: docs/extending/add_udw_tab.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        .enableReactPreset()
+        .enableSingleRuntimeChunk()
+    
+    // Put your config here.
+    
+    ibexaConfigManager.add({
+    ibexaConfig,
+    entryName: 'ibexa-admin-ui-udw-tabs-js',
+    newItems: [path.resolve(__dirname, './assets/js/image-tab/image.tab.module.js')],
+    });
```
Comment (dew326) [inline docs/extending/add_udw_tab.md:141]:
```suggestion
    ibexaConfigManager.add({
        ibexaConfig,
        entryName: 'ibexa-admin-ui-udw-tabs-js',
        newItems: [path.resolve(__dirname, './assets/js/image-tab/image.tab.module.js')],
    });
```

File: code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js [hunk 8]
```diff
@@ -1,28 +1,28 @@
...
+        <div className="m-image-tab">
+            <Tab isContentOnTheFlyDisabled={true} isSortSwitcherDisabled={true} isViewSwitcherDisabled={true}>
+                <ImagesList />
+            </Tab>
+        </div>
+    );
+};
 ibexa.addConfig(
-    'adminUiConfig.universalDiscoveryWidget.tabs',
+'adminUiConfig.universalDiscoveryWidget.tabs',
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js:16]:
```suggestion
    'adminUiConfig.universalDiscoveryWidget.tabs',
```

**Review by dew326 — CHANGES_REQUESTED** [review]

File: code_samples/back_office/udw/assets/js/image-tab/components/images.list.js [hunk 9]
```diff
@@ -76,8 +76,8 @@ const ImagesList = () => {
 
         return (
             <div {...attrs}>
-                <svg className="ibexa-icon">
-                    <use xlinkHref="/bundles/ibexaplatformicons/img/all-icons.svg#caret-next"></use>
+                <svg className="ez-icon">
```
Comment (dew326) [inline code_samples/back_office/udw/assets/js/image-tab/components/images.list.js:79]:
```suggestion
                <svg className="ibexa-icon">
```

File: docs/extending/add_udw_tab.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+    ibexaConfigManager.add({
+        ibexaConfig,
+        entryName: 'ibexa-admin-ui-udw-tabs-js',
+        newItems: [path.resolve(__dirname, './assets/js/image-tab/image.tab.module.js')],
+    });
+    
+    ibexaConfigManager.add({
+    ibexaConfig,
+    entryName: 'ibexa-admin-ui-layout-css',
+    });
```
Comment (dew326) [inline docs/extending/add_udw_tab.md:147]:
```suggestion
    ibexaConfigManager.add({
        ibexaConfig,
        entryName: 'ibexa-admin-ui-layout-css',
    });
```

File: docs/extending/add_udw_tab.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+    
+    // uncomment the two lines below, if you added a new entry (by Encore.addEntry() or Encore.addStyleEntry() method) to your own Encore configuration for your project
+    // const projectConfig = Encore.getWebpackConfig();
+    // module.exports = [ eZConfig, ...customConfigs, projectConfig ];
+    
+    // comment-out this line if you've uncommented the above lines
+    module.exports = [ ibexaConfig, ...customConfigs, projectConfig ];
```
Comment (dew326) [inline docs/extending/add_udw_tab.md:154]:
```suggestion
    module.exports = [ ibexaConfig, ...customConfigs ];
```

**Review by dew326 — APPROVED** [review]

File: code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js [hunk 12]
```diff
@@ -1,28 +1,28 @@
...
-            <div className="m-image-tab">
-                <Tab isContentOnTheFlyDisabled={true} isSortSwitcherDisabled={true} isViewSwitcherDisabled={true}>
-                    <ImagesList />
-                </Tab>
-            </div>
-        );
-    };
+
+import Tab from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/universal-discovery/components/tab/tab';
+import ImagesList from './components/image.list';
```
Comment (DominikaK) [inline code_samples/back_office/udw/assets/js/image-tab/image.tab.module.js:4]:
```suggestion
import ImagesList from './components/images.list';
```
This actually worked forme only as `images`

File: docs/extending/add_udw_tab.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+### Add css to webpack
+
+Finally, add css in `webpack.config.js`:
+
+```js
+ibexaConfigManager.add({
+    ibexaConfig,
+    entryName: 'ibexa-admin-ui-layout-css',
+    newItems: [path.resolve(__dirname, './assets/css/image.css'), path.resolve(__dirname, './assets/css/images.list.css')],
```
Comment (DominikaK) [inline docs/extending/add_udw_tab.md:113]:
```suggestion
    newItems: [path.resolve(__dirname, './assets/css/image.css'), path.resolve(__dirname, './assets/css/image.list.css')],
```

File: docs/extending/add_udw_tab.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+    // const projectConfig = Encore.getWebpackConfig();
+    // module.exports = [ eZConfig, ...customConfigs, projectConfig ];
+    
+    // comment-out this line if you've uncommented the above lines
+    module.exports = [ ibexaConfig, ...customConfigs ];
+    ```
+
+## Check results
+    
+In the Back Office go to **Content** -> **Content structure**. On the left panel, click **Browse**.
```
Comment (DominikaK) [inline docs/extending/add_udw_tab.md:159]: There is not more "Browse" in left panel, that's leftover from v3. The quickest way I can think of to get to UDW is to create an article and in Richtext field go to embed.

File: docs/extending/add_udw_tab.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+In the UDW a new **Images** tab appears, listing all images from the Repository.
+
+![Image tab in UDW](img/udw_image_tab.png)
+
+!!! tip
+
+    If you cannot see the results or encounter an error, clear the cache and reload the application. 
+    Remember, after any change of css/js file you should always run `yarn encore dev` in the terminal.
```
Comment (DominikaK) [inline docs/extending/add_udw_tab.md:167]:
```suggestion
    Remember, after any change of css/js files you should always run `yarn encore dev` in the terminal.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

File: docs/extending/add_udw_tab.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+ibexaConfigManager.add({
+    ibexaConfig,
+    entryName: 'ibexa-admin-ui-layout-css',
+    newItems: [path.resolve(__dirname, './assets/css/image.css'), path.resolve(__dirname, './assets/css/image.list.css')],
+});
+```
+
+## Check results
+    
+In the Back Office go to **Content** -> **Dashboard**. On the left panel, click the **Create content** button.
```
Comment (DominikaK) [inline docs/extending/add_udw_tab.md:119]:
```suggestion
In the Back Office go to **Content** -> **Dashboard**. On the top right, click the **Create content** button.
```

**Review by DominikaK — APPROVED** [review]

**Pull request merged**